### ISSUE TITLE
Crash to desktop after adding C# script to node then deleting, then readding+re-deleting a few times (3.2.4 rc5 mono)

### ISSUE THREAD

**Issue by rick551a** [godotengine/godot#46974]
**Godot version:**
Godot_v3.2.4-rc5_mono_win64

**OS/device including version:**
Win10 Nvidia 1650

**Issue description:**

Stumbled across this issue by adding ,then deleting then readding a c# sscript to a 3d node

**Steps to reproduce:**
1)Create a new blank project
2)Add a spatial 3D (dunno if this exact type of node is required, but this is how i stumbled on it)
3)Save project and make the node the default 'main' scene
4)add c# script to it. Save it close. Save project again.
5)run the project.
6)Click the node and remove the script. Save the Project
7)Move the c# script to the trash from the file editor.
8)run the project.
9)Repeat steps 4 to 8 a few times. It will crash on run eventually

**Minimal reproduction project:**

It will work on any new project. 
here is a pastebin of the mono  log
https://pastebin.com/MPiUfrSd

**Comment by Calinou** [comment]
@rick551a Can you reproduce this in 3.2.3?

**Comment by rick551a** [comment]
Yes, this issue is present, 100% reproducible, in 3.2.3 mono stable also. 
On windows7 it says "Godot has stopped working" before crash, on win10 it just crashes without warning.
Here is the event log error message:
'''
Faulting application name: Godot_v3.2.3-stable_mono_win64.exe, version: 3.2.3.0, time stamp: 0x00000000
Faulting module name: Godot_v3.2.3-stable_mono_win64.exe, version: 3.2.3.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x0000000002b63414
Faulting process id: 0xa40
'''

**Comment by 31** [comment]
Reproduces for me on Windows 10, 3.3-rc7. Also on a build of b076150b086a5001b190a9a20a425d1bc842fe21 (3.x tip).

I simplified the repro:

1. New project, make a 3d scene, save it.
2. **Create/attach a C# script to the node.**
3. Save.
4. Click "Build".
5. **Detach the script from the node.**
6. Save.
6. **Delete the script in the Godot file browser.**
7. Click "Build".
8. **Create/attach a C# script to the node.** (Any name--same as the first script or not.)
9. Save.
10. Click "Build"
11. Crash!

```
[0] GDMonoClass::get_nesting_class (c:\proj\godot\modules\mono\mono_gd\gd_mono_class.cpp:102)
[1] GDMonoClass::get_name_for_lookup (c:\proj\godot\modules\mono\mono_gd\gd_mono_class.cpp:90)
[2] CSharpLanguage::reload_assemblies (c:\proj\godot\modules\mono\csharp_script.cpp:802)
[3] _GodotSharp::_reload_assemblies (c:\proj\godot\modules\mono\mono_gd\gd_mono.cpp:1404)
[4] MethodBind1<_GodotSharp,bool>::call (c:\proj\godot\core\method_bind.gen.inc:867)
[5] Object::call (c:\proj\godot\core\object.cpp:919)
[6] MessageQueue::_call_function (c:\proj\godot\core\message_queue.cpp:250)
[7] MessageQueue::flush (c:\proj\godot\core\message_queue.cpp:299)
[8] SceneTree::idle (c:\proj\godot\scene\main\scene_tree.cpp:543)
[9] Main::iteration (c:\proj\godot\main\main.cpp:2123)
[10] OS_Windows::run (c:\proj\godot\platform\windows\os_windows.cpp:3440)
[11] widechar_main (c:\proj\godot\platform\windows\godot_windows.cpp:162)
[12] _main (c:\proj\godot\platform\windows\godot_windows.cpp:184)
[13] main (c:\proj\godot\platform\windows\godot_windows.cpp:196)
[14] __scrt_common_main_seh (d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[15] BaseThreadInitThunk

Exception thrown: read access violation.
**this** was nullptr.
```

Might be related, same method name on call stack: https://github.com/godotengine/godot/issues/46777.

**Comment by 31** [comment]
Repros on `master`, too. (f96392a2b5ee1ebd9c14c92bdcd24d25e8b7774b) If I add a null check specific to what's happening here, I can't get it to repro, and things *seem* to behave normally: PR https://github.com/godotengine/godot/pull/47691.

**Comment by akien-mga** [comment]
Fixed by #47691.
